app/queries/drug_stocks_query.rb

```diff
@@ -187,6 +185,7 @@ def drug_stocks_cache_key
     [
       "#{self.class.name}#drug_stocks",
       @facilities.map(&:id).sort,
+      @latest_drug_stocks.cache_key,
```

Comment:
@ssrihari @vkrmis This will correctly invalidate the cache when new drug stocks are created, so we don't do any `bust_cache` redirection to handle it. See https://bigbinary.com/blog/activerecord-relation-cache-key for some context on how this works.

Reply:
Thanks for doing this, @rsanheim. We had made a mental note to fix this later on. We recognise that the redirection was more a hack than a fix.